Sources/StreamChatUI/CommonViews/ContainerStackView.swift

```diff
@@ -113,17 +114,27 @@ public class ContainerStackView: UIView {
             respectsLayoutMarginsByView.removeValue(forKey: subview)
         }
 
+        hidingObserversByView[subview] = subview
+            .observe(\.isHidden, options: [.new]) { [weak self] (view, isHiddenChange) in
+                if isHiddenChange.newValue == true {
+                    self?.hideArrangedSubview(view)
+                } else {
+                    self?.showArrangedSubview(view)
+                }
+            }
+
```

Comment:
Great! 👍 I was thinking to also observe `superview` property. When the `superview` is changed the container is not responsible for this subview anymore and shouldn't track the `isHidden` field. It also will be super easy to move views between containers.

Reply:
In order to remove the subview from the container, we need to call `removeArrangedSubview`, which will remove the observer. Although you can use `removeFromSuperview` in a view inside the container, that is not expected, and it is a misuse of the Container, probably this should also be mentioned in the `removeArrangedSubview` documentation, I will add this 👍 But for me this is enough, I don't think it necessary to observe the `superview` property, or I'm missing something else?

Reply:
With `superview` observer it would be safe to do `myView.removeFromSuperview` or `anotherView.addSubview(myView)` or `anotherContainer.addArrangedSubview(myView)`. The idea was to have the insurance from misuse but on the other hand, this encourages misuse since nothing will get broken 😄 You are right, we don't need it 👍